### ISSUE TITLE
Add the corresponding criteria to each column of the training results.

### ISSUE THREAD

**Issue by SpongeBab** [ultralytics/yolov5#3271]
## 🚀 Feature
Add the corresponding criteria to each column of the training results.The first line should be the name of each metric.

## Motivation
When I check the results.txt file, I can't immediately know which Metric each column represents.

## Before:
```
     0/299      7.1G    0.1078     0.143         0    0.2508       106       640         0         0 2.545e-05 4.918e-06    0.1185    0.1189         0
     1/299        7G    0.0995    0.1547         0    0.2542       211       640   0.07547   0.02152  0.008825  0.002003   0.09631    0.1463         0
     2/299        7G   0.09501     0.156         0     0.251        93       640    0.1106   0.02552    0.0122   0.00238   0.09164    0.1464         0
     3/299        7G      0.09    0.1529         0    0.2429        66       640    0.1851   0.09165   0.03515  0.009394   0.08687    0.1477         0
     4/299        7G   0.08775     0.152         0    0.2397       137       640     0.182    0.1262   0.05595   0.01643   0.08404    0.1463         0
     5/299        7G   0.08626    0.1505         0    0.2368       168       640     0.249     0.113   0.06431   0.02068   0.08305    0.1449         0
     6/299        7G   0.08356    0.1492         0    0.2327       100       640    0.2565    0.1269   0.07044   0.02278   0.08133    0.1429         0
     7/299        7G   0.08208    0.1533         0    0.2354        90       640    0.2492   0.05537   0.03332  0.008882   0.08813      0.14         0
     8/299        7G   0.08109     0.147         0    0.2281        66       640    0.2294    0.1864    0.1131   0.03659   0.07858    0.1379         0
     9/299        7G   0.07937    0.1489         0    0.2283        94       640    0.2646    0.1156   0.07782   0.02503   0.08156    0.1349         0
    10/299        7G   0.07879    0.1446         0    0.2234        85       640    0.2573    0.1897    0.1458   0.04293   0.07931    0.1311         0
    11/299        7G   0.07859    0.1431         0    0.2217       130       640    0.2687    0.1725    0.1385   0.04729   0.07856    0.1326         0
    12/299        7G   0.07583     0.137         0    0.2128       178       640    0.2576     0.217    0.1736   0.06211   0.07786     0.128         0
    13/299        7G   0.07396    0.1371         0    0.2111       116       640    0.1777     0.214    0.1561   0.05492   0.07628    0.1288         0
    14/299        7G   0.07288    0.1401         0    0.2129       150       640     0.202    0.2257    0.1659   0.05788   0.07442    0.1285         0
......
```
## After
```
epoch      GPU    xxx            xxxx        xxx
 0/299      7.1G    0.1078     0.143         0    0.2508       106       640         0         0 2.545e-05 4.918e-06    0.1185    0.1189         0
```

**Comment by glenn-jocher** [comment]
@SpongeBab good news 😃! Your original issue may now be fixed ✅ in PR #4148. This PR updates results.txt to a new CSV format including column headers.

<img width="800" alt="Screenshot 2021-07-25 at 18 13 03" src="https://user-images.githubusercontent.com/26833433/126905881-6ef02b92-a89b-42ee-9261-df46fc6cd06c.png">
<img width="800" alt="Screenshot 2021-07-25 at 18 13 03" src="https://user-images.githubusercontent.com/26833433/126906780-8c5e2990-6116-4de6-b78a-367244a33ccf.png">

To receive this update:
- **[Git](https://github.com/ultralytics/yolov5)** – `git pull` from within your `yolov5/` directory or `git clone https://github.com/ultralytics/yolov5` again
- **[PyTorch Hub](https://pytorch.org/hub/ultralytics_yolov5/)** – Force-reload with `model = torch.hub.load('ultralytics/yolov5', 'yolov5s', force_reload=True)`
- **[Notebooks](https://github.com/ultralytics/yolov5/blob/master/tutorial.ipynb)** – View updated notebooks  <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **[Docker](https://hub.docker.com/r/ultralytics/yolov5)** – `sudo docker pull ultralytics/yolov5:latest` to update your image <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>

Thank you for spotting this issue and informing us of the problem. Please let us know if this update resolves the issue for you, and feel free to inform us of any other **issues** you discover or **feature requests** that come to mind. Happy trainings with YOLOv5 🚀!
